src/SelfServe/Documentation/README.md

```diff
@@ -0,0 +1,296 @@
+`<FEATURE_NAME>.rp.ts`
+
+#### Example
+[SelfServeExample.rp.ts](https://github.com/Azure/cosmos-explorer/blob/master/src/SelfServe/Example/SelfServeExample.rp.ts)
+
+#### Description
+The RP file will host the REST calls needed for the initialize, save and refresh functions. This decouples the code that builds the UI and the code that makes REST calls.
+
+The `userContext` object exposes the following Azure Cosmos DB databaseAccount information
+* `userContext.subscriptionId` - The subscription id
+* `userContext.resourceGroup` - The resource group name 
+* `userContext.databaseAccount.name` - The database account name
+
+`armRequestWithoutPolling` function can be used to make the ARM api call.
+
+Your `FeatureName.rp.ts` file can look like the following.
+
+```ts
+import { userContext } from "../../UserContext";
+import { armRequestWithoutPolling } from "../../Utils/arm/request";
+import { configContext } from "../../ConfigContext";
+
+const apiVersion = "2020-06-01-preview";
+
+export const saveData = async (properties: PropertiesToSave): Promise<void> => {
+  const path = `/subscriptions/${userContext.subscriptionId}/resourceGroups/${userContext.resourceGroup}/providers/Microsoft.DocumentDB/databaseAccounts/${userContext.databaseAccount.name}/<REST_OF_THE_PATH>`
+  const body = {
+      data : properties
+  }
+  const armRequestResult = await armRequestWithoutPolling({
+    host: configContext.ARM_ENDPOINT,
+    path,
+    method: "PUT",
+    apiVersion,
+    body,
+  });
+};
+
+```
+
+### 4. Class file
+
+#### Naming Convention
+`<FEATURE_NAME>.tsx`
+
+#### Example
+[SelfServeExample.tsx](https://github.com/Azure/cosmos-explorer/blob/master/src/SelfServe/Example/SelfServeExample.tsx)
+
+#### Description
+This file will contain the actual code that is translated into the UI component by the Self Serve framework.
+* Each Self Serve class
+  * Needs to extends the [SelfServeBase](../classes/selfserve_selfservetypes.selfservebaseclass.html) class.
+  * Needs to have the [@IsDisplayable()](./selfserve_decorators.html#isdisplayable) decorator to tell the compiler that UI needs to be generated from this class.
+  * Needs to define an [initialize()](../classes/selfserve_selfservetypes.selfservebaseclass.html#initialize) function, to set default values for the inputs.
+  * Needs to define an [onSave()](../classes/selfserve_selfservetypes.selfservebaseclass.html#onsave) function, a callback for when the save button is clicked.
+  * Needs to define an [onRefresh()](../classes/selfserve_selfservetypes.selfservebaseclass.html#onrefresh) function, a callback for when the refresh button is clicked.
+  * Can have an optional [@RefreshOptions()](./selfserve_decorators.html#refreshoptions) decorator that determines how often the auto refresh of the UI component should take place.
+
+* Each property of the Self Serve class that needs to be translated to a UI element
+  * Needs to have a [@Values()](./selfserve_decorators.html#values) decorator.
```

Comment:
For every UI element needed in your feature, create a property in the Self Serve class. Each of these properties...